src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -107,7 +192,7 @@ internal static class ParameterHelpers
                     mustBeLastParameter.GetLocation());
             }
 
-            ImmutableArray<ParameterSymbol> parameters = builder.ToImmutableAndFree();
+            ImmutableArray<TParameterSymbol> parameters = builder.ToImmutableAndFree();
 
             var methodOwner = owner as MethodSymbol;
```

Comment:
>var methodOwner = owner as MethodSymbol; [](start = 12, length = 40)

It feels like ```if (!parsingFunctionPointer)``` should be moved right above this line. All locals are calculated solely for the purpose of ValidateParameterNameConflicts call. The ```Debug.Assert(methodOwner?.MethodKind != MethodKind.LambdaMethod);``` will definitely be true for a function pointer, and, it feels like, the assert is tied to the following ```IDS_FeatureNameShadowingInNestedFunctions``` feature check anyway.